jerry-core/ecma/builtin-objects/ecma-builtin-math.c

```diff
@@ -49,596 +84,289 @@
- *
- * See also:
- *          ECMA-262 v5, 15.8.2.5
+ *          ECMA-262 v5, 15.8.2.11
+ *          ECMA-262 v5, 15.8.2.12
  *
  * @return ecma value
  *         Returned value must be freed with ecma_free_value.
  */
 static ecma_value_t
-ecma_builtin_math_object_atan2 (ecma_value_t this_arg, /**< 'this' argument */
-                                ecma_value_t arg1, /**< first routine's argument */
-                                ecma_value_t arg2) /**< second routine's argument */
+ecma_builtin_math_object_max_min (bool is_max, /**< 'max' or 'min' operation */
+                                  const ecma_value_t *arg, /**< arguments list */
+                                  ecma_length_t args_number) /**< number of arguments */
 {
-  JERRY_UNUSED (this_arg);
-  ecma_value_t ret_value = ecma_make_simple_value (ECMA_SIMPLE_VALUE_EMPTY);
+  ecma_number_t result_num = ecma_number_make_infinity (is_max);
 
-  ECMA_OP_TO_NUMBER_TRY_CATCH (x, arg1, ret_value);
-  ECMA_OP_TO_NUMBER_TRY_CATCH (y, arg2, ret_value);
-
-  ret_value = ecma_make_number_value (DOUBLE_TO_ECMA_NUMBER_T (atan2 (x, y)));
+  while (args_number > 0)
+  {
+    ecma_number_t arg_num;
 
-  ECMA_OP_TO_NUMBER_FINALIZE (y);
-  ECMA_OP_TO_NUMBER_FINALIZE (x);
-  return ret_value;
-} /* ecma_builtin_math_object_atan2 */
+    if (ecma_is_value_number (*arg))
+    {
+      arg_num = ecma_get_number_from_value (*arg);
+    }
+    else
+    {
+      ecma_value_t value = ecma_op_to_number (*arg);
 
-/**
- * The Math object's 'ceil' routine
- *
- * See also:
- *          ECMA-262 v5, 15.8.2.6
- *
- * @return ecma value
- *         Returned value must be freed with ecma_free_value.
- */
-static ecma_value_t
-ecma_builtin_math_object_ceil (ecma_value_t this_arg, /**< 'this' argument */
-                               ecma_value_t arg) /**< routine's argument */
-{
-  JERRY_UNUSED (this_arg);
-  ecma_value_t ret_value = ecma_make_simple_value (ECMA_SIMPLE_VALUE_EMPTY);
+      if (ECMA_IS_VALUE_ERROR (value))
+      {
+        return value;
+      }
 
-  ECMA_OP_TO_NUMBER_TRY_CATCH (arg_num, arg, ret_value);
+      arg_num = ecma_get_number_from_value (value);
 
-  ret_value = ecma_make_number_value (DOUBLE_TO_ECMA_NUMBER_T (ceil (arg_num)));
+      ecma_fast_free_value (value);
+    }
 
-  ECMA_OP_TO_NUMBER_FINALIZE (arg_num);
-  return ret_value;
-} /* ecma_builtin_math_object_ceil */
+    if (unlikely (ecma_number_is_nan (arg_num)))
+    {
+      result_num = arg_num;
+      break;
+    }
 
-/**
- * The Math object's 'cos' routine
- *
- * See also:
- *          ECMA-262 v5, 15.8.2.7
- *
- * @return ecma value
- *         Returned value must be freed with ecma_free_value.
- */
-static ecma_value_t
-ecma_builtin_math_object_cos (ecma_value_t this_arg, /**< 'this' argument */
-                              ecma_value_t arg) /**< routine's argument */
-{
-  JERRY_UNUSED (this_arg);
-  ecma_value_t ret_value = ecma_make_simple_value (ECMA_SIMPLE_VALUE_EMPTY);
+     /* both numbers are zeroes */
```

Comment:
Comment should start with uppercase letter and end with full stop. Though honestly I think you can just drop the comment :)